### ISSUE TITLE
Can't open zipped shapefile stored locally with latest version 1.7.2

### ISSUE THREAD

**Issue by polin924** [Toblerity/Fiona#418]
I have one zipped shapefile 'test.zip' stored locally ( /tmp/dir1/test.zip)
Unzipping the zipped shapefile, the structure is as below:
```
   test/a.shx
   test/a.shp
   test/a.prj 
   test/a.dbf.
```

I followed the tutorial an did like this:
```
    with fiona.open(  '/test/test.dbf',   vfs='zip:///tmp/dir1/test.zip') as c:
        print(len(c))
```
Fiona throws fiona.errors.FionaValueError: No dataset found at path '/vsizip/tmp/dir1/test.zip/test/a.dbf'  using drivers: *

I tried the older version (1.7.1). It works. 
Is it a bug  or what I did was wrong ?

**Comment by geowurster** [comment]
@polin924 Looks like you're trying to open `/test/test.dbf` instead of `/test/a.shp`.  Let me know if that doesn't work and I'll reopen.

**Comment by polin924** [comment]
Hi Kevin,

Sorry that is a typo.
In my code, I actually tried to open ' /test/a.dbf' instead of
/test/test.dbf.
I was able to do so in version 1.7.1 but not 1.7.2

thanks

**Comment by polin924** [comment]
Sorry that is a typo.
What I did really is as below:
```
with fiona.open(  '/test/a.dbf',   vfs='zip:///tmp/dir1/test.zip') as c:
        print(len(c))
```

**Comment by geowurster** [comment]
Yep, looks like a bug around translating the VFS schema to a `/vsizip/` string:

```
Traceback (most recent call last):
  File "buh.py", line 3, in <module>
    with fiona.open('/test/a.shp',   vfs='zip:///tmp/dir1/test.zip') as c:
  File "/Users/wursterk/Library/Python/2.7/lib/python/site-packages/fiona/__init__.py", line 169, in open
    enabled_drivers=enabled_drivers)
  File "/Users/wursterk/Library/Python/2.7/lib/python/site-packages/fiona/collection.py", line 148, in __init__
    self.session.start(self)
  File "fiona/ogrext.pyx", line 444, in fiona.ogrext.Session.start (fiona/ogrext2.c:9237)
fiona.errors.FionaValueError: No dataset found at path '/vsizip/tmp/dir1/test.zip/test/a.shp' using drivers: *
```

Based on the exception, the `/vsizip/` path to the zip file is switched from absolute to relative.

**Comment by geowurster** [comment]
@polin924 Actually I think this may be broken due to a packaging issue on our end.  Could you answer the following questions in detail:

1. How did you install Fiona?
2. What version of GDAL are you running?  You can do this with: `$ python -c "import fiona; print(fiona.__gdal_version__)"`.
3. What happens when you do: `$ gdal-config --version`?
3. What happens when you install Fiona with the command below and then run the same test?  This will install Fiona in a slightly different manner and may take a few minutes.

```console
$ pip install git+https://github.com/toblerity/Fiona --upgrade --no-deps
```

**Comment by polin924** [comment]
1. I installed fiona through pip 
   ```pip3 install fiona ```
2. I am using python3.5
3.  On my computer, it shows 2.1.2 after I did
    ```python3 -c "import fiona; print(fiona.__gdal_version__)" ```
4. After I did `gdal-config --version` I got 1.11.2.
5. I was trying to do ```pip3 install git+https://github.com/toblerity/Fiona --upgrade --no-deps 
```
since I am using python3
but I got installation error. Therefore I tried 
```pip install git+https://github.com/toblerity/Fiona --upgrade --no-deps ```
and then tried running the same test in python2.7.10. It works.

**Comment by geowurster** [comment]
@polin924 What version of Python 3?  Can you share the error from `$ pip3 install git+https://github.com/toblerity/Fiona --upgrade --no-deps`.

**Comment by geowurster** [comment]
@polin924 Can you share the error from `$ pip3 install git+https://github.com/toblerity/Fiona --upgrade --no-deps`?

**Comment by geowurster** [comment]
@polin924 I figured it out.  The bug you discovered is a regression that was introduced before `1.7.2` was released, so VFS URL's are partially broken in the latest release.  The bug has already been fixed in https://github.com/Toblerity/Fiona/commit/699ece9c9c3f3d9a934d2daa895c4c98df98be2b.

We will have to wait for @sgillies (who is out this week) to decide on a a release plan to get the fix into PyPI.  In the meantime I would recommend using `1.7.1.post1` unless you need the [changes introduced in `1.7.2`](https://github.com/Toblerity/Fiona/blob/master/CHANGES.txt#L6), in which case you can install directly from the commit responsible for fixing the regression with:

```console
$ pip install git+git://github.com/toblerity/Fiona.git@699ece9c9c3f3d9a934d2daa895c4c98df98be2b
```

**Comment by sgillies** [comment]
Fixed in 1.7.3.
